internal/github/checkStalePRs.go

```diff
@@ -0,0 +1,68 @@
+package github
+
+import (
+	"context"
+	"fmt"
+	"log"
+	"time"
+
+	"github.com/google/go-github/v60/github"
+
+	"github.com/chia-network/github-bot/internal/config"
+)
+
+// CheckStalePRs will return a list of PR URLs that have not been updated in the last 7 days by internal team members.
+func CheckStalePRs(githubClient *github.Client, internalTeam string, cfg config.CheckStalePending) ([]string, error) {
+	var stalePRUrls []string
+	cutoffDate := time.Now().AddDate(0, 0, -7) // 7 days ago
+	teamMembers, err := GetTeamMemberList(githubClient, internalTeam)
+	if err != nil {
+		return nil, err
+	}
+	communityPRs, err := FindCommunityPRs(cfg.CheckStalePending, teamMembers, githubClient)
+	if err != nil {
+		return nil, err
+	}
+
+	for _, pr := range communityPRs {
+		stale, err := isStale(githubClient, pr, teamMembers, cutoffDate) // Handle both returned values
+		if err != nil {
+			log.Printf("Error checking if PR is stale: %v", err) // Log or handle the error
+			continue                                             // Skip this PR or handle the error appropriately
+		}
+		if stale {
+			stalePRUrls = append(stalePRUrls, pr.GetHTMLURL()) // Append if PR is confirmed stale
+		}
+	}
+	return stalePRUrls, nil
+}
+
+// Checks if a PR is stale based on the last update from team members
+func isStale(githubClient *github.Client, pr *github.PullRequest, teamMembers map[string]bool, cutoffDate time.Time) (bool, error) {
+	// Set up a context with a timeout to control all operations within this function
+	ctx, cancel := context.WithTimeout(context.Background(), 30*time.Second)
```

Comment:
I believe this is limiting this whole function to 30 seconds? What if there are a lot of pages to fetch and it takes longer than that?

Reply:
Changed to 300s

Reply:
> Changed to 300s

The point I was getting at is that a timeout for a loop of unknown length is probably not the route we want to take. If your set on keeping a context here, then it should likely just be a context for each request, if the requests don't support timeouts more directly